app/osparc/osparc.py

```diff
@@ -152,10 +152,10 @@ def check_simulation(data):
         solver_version = data["solver"]["version"]
         status = solvers_api.inspect_job(solver_name, solver_version, job_id)
 
-        if status.progress == 100:
+        if status.state in {"SUCCESS", "FAILED"}:
             # The simulation has completed, but was it successful?
 
-            if status.state != "SUCCESS":
+            if status.state == "FAILED":
                 raise SimulationException("the simulation failed")
```

Comment:
Seems a bit odd to do it this way. To me the logic is better if it is just:
```
if status.state == "FAILED":
    raise SimulationException("the simulation failed")

elif status.state == "SUCCESS":
    # Do success stuff ...
else:
    raise SimulationException("something else happened")
```
I don't know the API if it can return anything other than `SUCCESS` or `FAILED`.  If those are the only possible outcomes then I would simplify the logic.

Reply:
Yes, for our purpose, they are the only two relevant states. Otherwise, yes, agreed about your suggestion. I guess I was a bit slack by just reusing the previous logic! Anyway: 64b1e9b.